### PR TITLE
Update JSON readme

### PR DESCRIPTION
**Pull request by mristin**
We update the readme to describe the up-to-date JSON serialization.

### CODE REVIEW

**Review by BirgitBoss — COMMENTED** [review]
seems no consensus yet how to deal with xsd types and strings
+
see comments, especially 
- question w.r.t. aggregations of cardinality 0..1
- list exceptions from rule to add "s" for cardinality 0..*
- explain what changed because of gernations (last paragraph, soemthing with enumerations e.g.)

File: schemas/json/README.md [hunk 1]
```diff
@@ -1,37 +1,177 @@
 # JSON
-In the following repository an overview of the main concepts of the Asset Administration Shell [JSON](https://tools.ietf.org/html/rfc8259) serialization is presented. For import and export scenarios the metamodel of an Asset Administration Shell needs to be serialized. A serialization format is JSON (JavaScript Object Notation). The JSON files follow the schema version [2019-09](https://json-schema.org/specification-links.html#2019-09-formerly-known-as-draft-8).
 
-## Top-Level Structure of Asset Administration Shell Environment
-One serialization describes one AAS environment, that is, a collection of Administration Shells. The root element of the Asset Administration Shell environment has 5 aggregations. AssetAdministrationShells, Assets, Submodels, ConceptDescriptions and Data Specifications. For each identifiable class, one aggregation is provided, as required by rule 3.
+[JavaScript Object Notation (JSON)] is a popular serialization format.
+Beside XML, it is an  "official" serialization format for models of Asset Administration Shells (AAS).
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:4]: done

File: schemas/json/README.md [hunk 2]
```diff
@@ -1,37 +1,177 @@
...
+### Cardinality
+Optional attributes, *i.e.*, the attributes with the cardinality ``0..1``, are modeled as [non-required properties].
+
+[non-required properties]: https://json-schema.org/understanding-json-schema/reference/object.html#required-properties
+
+Attributes describing aggregations, *i.e.*, the attributes with the cardinality ``0..1`` or ``1..*``, are modeled as [JSON arrays].
+
+[JSON arrays]: https://json-schema.org/understanding-json-schema/reference/array.html
+
+For better readability, we diverge from the meta-model and spell most properties in plural.
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:54]: The properties could be called just as well in singular form. There is no compelling need to use singular or plural form, except readability.

File: schemas/json/README.md [hunk 3]
```diff
@@ -1,37 +1,177 @@
...
 
+### Classes 🠒 JSON definitions
+For each class of the [AAS meta-model], we provide [a definition] in the JSON schema.
+The instances of the classes, abstract and concrete alike, are modeled as [JSON objects].
 
+[a definition]: https://json-schema.org/understanding-json-schema/structuring.html#defs
+[JSON objects]: https://json-schema.org/understanding-json-schema/reference/object.html
+
+### Attributes  🠒 JSON properties
+The attributes of classes defined in the meta-model correspond directly to [JSON properties].
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:41]: JSON is an object notation. Attributes map directly to properties. Whether it is an association or not, JSON does not care.

File: schemas/json/README.md [hunk 4]
```diff
@@ -1,37 +1,177 @@
...
+[non-required properties]: https://json-schema.org/understanding-json-schema/reference/object.html#required-properties
+
+Attributes describing aggregations, *i.e.*, the attributes with the cardinality ``0..1`` or ``1..*``, are modeled as [JSON arrays].
+
+[JSON arrays]: https://json-schema.org/understanding-json-schema/reference/array.html
+
+For better readability, we diverge from the meta-model and spell most properties in plural.
+For example, ``submodelElements`` instead of ``submodelElement`` in case of [Submodel] class.
+
+If plural form made no sense for an attribute, we kept it as-is (*e.g.*, `isCaseOf`).
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:57]: added a reference to aas-core-meta

File: schemas/json/README.md [hunk 5]
```diff
@@ -1,37 +1,177 @@
...
+The attributes of classes defined in the meta-model correspond directly to [JSON properties].
+
+[JSON properties]: https://json-schema.org/understanding-json-schema/reference/object.html#properties
+
+### Cardinality
+Optional attributes, *i.e.*, the attributes with the cardinality ``0..1``, are modeled as [non-required properties].
+
+[non-required properties]: https://json-schema.org/understanding-json-schema/reference/object.html#required-properties
+
+Attributes describing aggregations, *i.e.*, the attributes with the cardinality ``0..1`` or ``1..*``, are modeled as [JSON arrays].
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:50]: This is a typo in my text. Fixed.

File: schemas/json/README.md [hunk 6]
```diff
@@ -1,37 +1,177 @@
...
+### Why only JSON strings?
+Let us briefly elaborate at least two reasons why the mapping XSD types 🠒 JSON types is not possible, since the readers are often confused why numeric and boolean values are represented as strings.
+
+First, a [JSON number] can only be a concrete numerical value where special cases such as `INF` and `NaN` are not representable in JSON, whereas these values are completely valid instances of [xs:decimal].
+
+[xs:decimal]: https://www.w3.org/TR/xmlschema11-2/#decimal
+
+Second, the [lexical space] of XSD values is much larger than the lexical space of JSON numbers and booleans.
+For example, both `1` and `true` are valid lexical representations of a true value.
+In contrast, JSON recognizes only `true`.
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:84]: Added a reference to the next section. Only injection is possible, not bijection.

File: schemas/json/README.md [hunk 7]
```diff
@@ -1,37 +1,177 @@
...
+[Range]: https://www.plattform-i40.de/IP/Redaktion/DE/Downloads/Publikation/Details_of_the_Asset_Administration_Shell_Part1_V3.pdf?__blob=publicationFile&v=10#page=76
+[JSON number]: https://www.rfc-editor.org/rfc/rfc4627#section-2.4
+[JSON boolean]: https://json-schema.org/understanding-json-schema/reference/boolean.html
+[XSD types]: https://www.w3.org/TR/xmlschema11-2
+[5.7.12.1 Predefined Simple Data Types]: https://www.plattform-i40.de/IP/Redaktion/DE/Downloads/Publikation/Details_of_the_Asset_Administration_Shell_Part1_V3.pdf?__blob=publicationFile&v=10#page=95 
+
+### Why only JSON strings?
+Let us briefly elaborate at least two reasons why the mapping XSD types 🠒 JSON types is not possible, since the readers are often confused why numeric and boolean values are represented as strings.
+
+First, a [JSON number] can only be a concrete numerical value where special cases such as `INF` and `NaN` are not representable in JSON, whereas these values are completely valid instances of [xs:decimal].
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:78]: No, it is indeed a general rule. As I explain, only injection JSON -> XML is possible, not the bijection.

Whether the value is given as JSON number or string plays no role in enforcing constraints. The code which enforces constraints needs to appropriately interpret the JSON string values.

File: schemas/json/README.md [hunk 8]
```diff
@@ -1,37 +1,177 @@
...
+### ... to Automatic Generation
+We eventually moved over to generate the serialization schemas based on a single-point-of-truth.
+The current source is [aas-core-meta], a domain-specific Python representation of the meta-model.
+The schemas are generated using [aas-core-codegen], a translating tool which transpiles aas-core-meta into different schemas and other targets such as SDKs.
+
+[aas-core-meta]: https://github.com/aas-core-works/aas-core-meta
+[aas-core-codegen]: https://github.com/aas-core-works/aas-core-codegen
+
+While this approach reduced the rate of errors significantly, it also imposed certain limits on our schema design.
+For example, the classes and enumerations are now programmatically mapped to JSON definitions, allowing for no exceptions.
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:176]: A lot of enumerations were in-lined. Won't fix as this takes too much time and adds little to the content.

File: schemas/json/README.md [hunk 9]
```diff
@@ -1,37 +1,177 @@
...
+[5.7.12.1 Predefined Simple Data Types]: https://www.plattform-i40.de/IP/Redaktion/DE/Downloads/Publikation/Details_of_the_Asset_Administration_Shell_Part1_V3.pdf?__blob=publicationFile&v=10#page=95 
+
+### Why only JSON strings?
+Let us briefly elaborate at least two reasons why the mapping XSD types 🠒 JSON types is not possible, since the readers are often confused why numeric and boolean values are represented as strings.
+
+First, a [JSON number] can only be a concrete numerical value where special cases such as `INF` and `NaN` are not representable in JSON, whereas these values are completely valid instances of [xs:decimal].
+
+[xs:decimal]: https://www.w3.org/TR/xmlschema11-2/#decimal
+
+Second, the [lexical space] of XSD values is much larger than the lexical space of JSON numbers and booleans.
```
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:82]: The XSD type is defined in `valueType` attribute. Please let me know which part of the document I need to fix to clarify even more that the bijection XML <-> JSON is not possible.

[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:82]: Let's clarify this issue over email.

File: schemas/json/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (BirgitBoss) [reply schemas/json/README.md:176]: This was a question not a change request. It is still unclear to me how it was before and how it is now. So you are saying: In-lined enumerations are not used any longer in the JSON schema? Is it about 

grafik
? (https://stackoverflow.com/questions/30924271/correct-way-to-define-array-of-enums-in-json-schema)

[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:176]: In-lining would be A.

What we do: we introduce a definition with type `string` and `enum` constraint, and reference it.

See:
https://github.com/admin-shell-io/aas-specs/blob/7ec3e0807bd4a528674a678d1a0d581c45a747ff/schemas/json/aas.json#L105

and

https://github.com/admin-shell-io/aas-specs/blob/7ec3e0807bd4a528674a678d1a0d581c45a747ff/schemas/json/aas.json#L125

File: schemas/json/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (BirgitBoss) [reply schemas/json/README.md:41]: Yes, but UML does. And on the left side it is UML of AAS metamodel.

[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:41]: Fixed. Used the term UML property from https://www.uml-diagrams.org/property.html

File: schemas/json/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:54]: Re-phrased as difference between UML and programming code.

File: schemas/json/README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (mristin) [reply schemas/json/README.md:82]: This rule needs to be re-written as JSON booleans and numbers should be used.

**Pull request merged**